specification/vision/ImageAnalysis/models.tsp

```diff
@@ -0,0 +1,324 @@
+
+  @doc("The model used for the analysis")
+  metadata: ImageMetadata;
+
+  @doc("The model used for the analysis")
+  modelVersion: string;
+
+  @doc("A list of objects for the image.")
+  objectsResult?: ObjectsResult;
+
+  @doc("A list of people for the image.")
+  peopleResult?: PeopleResult;
+
+  @doc("A readResult for the image.")
+  readResult?: ReadResult;
+
+  @doc("A list of regions for the image.")
+  smartCropsResult?: SmartCropsResult;
+
+  @doc("A list of tags for the image.")
+  tagsResult?: TagsResult;
+}
+
+@doc("The image metadata information such as height and width.")
+model ImageMetadata {
+  @doc("The height of the image in pixels.")
+  height: int32;
+
+  @doc("The width of the image in pixels.")
+  width: int32;
+}
+
+@doc("Describes the result of image analysis using a custom model.")
+model CustomModelResult {
+  @doc("The list of predicted objects.")
+  objectsResult: ObjectsResult;
+
+  @doc("The list of predicted tags.")
+  tagsResult: TagsResult;
+}
+
+@doc("Describes detected objects in an image.")
+model ObjectsResult {
+  @doc("An array of detected objects.")
+  values: Array<DetectedObject>;
+}
+
+@doc("An object describing whether the image contains people.")
+model PeopleResult {
+  @doc("An array of detected people.")
+  values: Array<DetectedPerson>;
+}
+
+@doc("The results of an Read operation.")
+model ReadResult {
+  @doc("Concatenate string representation of all textual and visual elements in reading order.")
+  content: string;
+
+  @doc("A list of analyzed pages.")
+  pages: Array<DocumentPage>;
+
+  @doc("The method used to compute string offset and length, possible values include: 'textElements', 'unicodeCodePoint', 'utf16CodeUnit' etc.")
+  stringIndexType: string;
+
+  @doc("Extracted font styles.")
+  styles: Array<DocumentStyle>;
+
+  @doc("The model used to generate the Read result.")
+  modelVersion: string;
+}
+
+@doc("Smart cropping result.")
+model SmartCropsResult {
+  @doc("Recommended regions for cropping the image.")
+  values: Array<CropRegion>;
+}
+
+@doc("An entity observation in the image, along with the confidence score.")
+model DetectedTag {
+  @doc("The level of confidence that the entity was observed.")
+  confidence: float32;
+
+  @doc("Name of the entity.")
+  name: string;
+}
+
+@doc("A list of tags with confidence level.")
+model TagsResult {
+  @doc("A list of tags with confidence level.")
+  values: Array<DetectedTag>;
+}
+
+@doc("The visual features requested: tags, objects, caption, denseCaptions, read, smartCrops, people. This parameter needs to be specified if the parameter \"model-name\" is not specified.")
+enum VisualFeatures {
+  @doc("Tags")
+  tags,
+
+  @doc("Caption")
+  caption,
+
+  @doc("DenseCaptions")
+  denseCaptions,
+
+  @doc("Objects")
+  objects,
+
+  @doc("Read")
+  read,
+
+  @doc("SmartCrops")
+  smartCrops,
+
+  @doc("People")
+  people,
+}
+
+@doc("The segmentation mode requested.")
+enum segmentationMode {
```

Comment:
nit, PascalCase for model/enum name, for consistency in tsp source

Reply:
@weidongxu-microsoft Good catch, thanks! Fixed.